src/internal/connector/exchange/exchange_data_collection.go

```diff
@@ -31,8 +34,9 @@ var (
 )
 
 const (
-	collectionChannelBufferSize = 1000
-	numberOfRetries             = 4
+	collectionChannelBufferSize  = 1000
+	numberOfRetries              = 4
+	urlPrefetchChannelBufferSize = 50
```

Comment:
@meain - see https://learn.microsoft.com/en-us/graph/throttling-limits#outlook-service-limits. It seems like we shouldn't have > 4 concurrent requests.